xla/pjrt/pjrt_stream_executor_client.cc

```diff
@@ -1255,14 +1259,43 @@ Status PjRtStreamExecutorDevice::TransferFromOutfeed(
       local_device->local_hardware_id().value(), literal);
 }
 
+void PjRtStreamExecutorClient::SetMemorySpaces(PjRtDevice* device) {
+  if (platform_name() == "cuda") {
+    device_space_ = std::make_unique<DeviceMemorySpace>(
+        static_cast<int>(Layout::kDefaultMemorySpace), device);
```

Comment:
The id of a memory space is an unique id that identifies a unique memory space.

The semantics we follow now is that each device will have independent memory spaces of a same kind.